apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/db/model/OptionDAOTest.java

```diff
@@ -0,0 +1,136 @@
+package com.apollocurrency.aplwallet.apl.core.db.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.util.Collections;
+
+import com.apollocurrency.aplwallet.apl.core.app.BlockDaoImpl;
+import com.apollocurrency.aplwallet.apl.core.app.BlockchainImpl;
+import com.apollocurrency.aplwallet.apl.core.app.DatabaseManager;
+import com.apollocurrency.aplwallet.apl.core.app.EpochTime;
+import com.apollocurrency.aplwallet.apl.core.app.TransactionDaoImpl;
+import com.apollocurrency.aplwallet.apl.core.chainid.BlockchainConfig;
+import com.apollocurrency.aplwallet.apl.core.db.TransactionalDataSource;
+import com.apollocurrency.aplwallet.apl.util.Constants;
+import com.apollocurrency.aplwallet.apl.util.NtpTime;
+import com.apollocurrency.aplwallet.apl.util.env.config.PropertiesConfigLoader;
+import com.apollocurrency.aplwallet.apl.util.env.dirprovider.ConfigDirProvider;
+import com.apollocurrency.aplwallet.apl.util.env.dirprovider.ConfigDirProviderFactory;
+import com.apollocurrency.aplwallet.apl.util.injectable.DbConfig;
+import com.apollocurrency.aplwallet.apl.util.injectable.DbProperties;
+import com.apollocurrency.aplwallet.apl.util.injectable.PropertiesHolder;
+import org.jboss.weld.junit5.WeldInitiator;
+import org.jboss.weld.junit5.WeldSetup;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class OptionDAOTest {
+
+    private static String BASE_SUB_DIR = "unit-test-db";
+
+    @WeldSetup
+    public WeldInitiator weld = WeldInitiator.from(DbProperties.class, NtpTime.class,
+            PropertiesConfigLoader.class,
+            PropertiesHolder.class, BlockchainConfig.class, BlockchainImpl.class, DbConfig.class,
+            EpochTime.class, BlockDaoImpl.class, TransactionDaoImpl.class,
+            TransactionalDataSource.class, DatabaseManager.class)
+            .build();
+
+    private static Path pathToDb;
+    private static PropertiesHolder propertiesHolder;
+    private static DbProperties baseDbProperties;
+    private static DatabaseManager databaseManager;
+    private static OptionDAO optionDAO;
+
+    @BeforeAll
+    static void setUpAll() {
+        ConfigDirProvider configDirProvider = new ConfigDirProviderFactory().getInstance(false, Constants.APPLICATION_DIR_NAME);
+        String workingDir = System.getProperty("user.dir");
+        pathToDb = FileSystems.getDefault().getPath(workingDir + File.separator  + BASE_SUB_DIR);
+        PropertiesConfigLoader propertiesLoader = new PropertiesConfigLoader(
```

Comment:
Mock properties holder and do not init propertiesConfigLoader for tests

Reply:
Let me try after full merge.